includes/class-wc-subscriptions-change-address-via-checkout-handler.php

```diff
@@ -0,0 +1,529 @@
+<?php
+use Automattic\WooCommerce\StoreApi\Exceptions\RouteException;
+
+/**
+ * Handles the change address via checkout flow.
+ *
+ * @package WooCommerce Subscriptions
+ * @since 5.9.0
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+class WC_Subscriptions_Change_Address_Via_Checkout_Handler {
+
+
+
+	/**
+	 * The cart class instance responsible for handling the subscription through the cart.
+	 *
+	 * @var
+	 */
+	private static $edit_subscription_cart_instance;
+
+	/**
+	 * Initializes the class and hooks.
+	 */
+	public static function init() {
+		add_action( 'wp_loaded', array( __CLASS__, 'maybe_add_subscription_to_cart_for_address_change' ) );
+		add_action( 'woocommerce_checkout_before_customer_details', array( __CLASS__, 'maybe_add_hidden_input_for_address_change' ) );
+		add_filter( 'woocommerce_update_order_review_fragments', array( __CLASS__, 'maybe_modify_checkout_template_for_address_change' ) );
+		add_action( 'woocommerce_checkout_process', array( __CLASS__, 'process_address_change' ) );
+		add_action( 'template_redirect', array( __CLASS__, 'maybe_navigated_away_from_change_address_page' ) );
+		add_filter( 'woocommerce_cart_needs_payment', array( __CLASS__, 'cart_requires_payment' ) );
+		add_filter( 'woocommerce_add_cart_item_data', array( __CLASS__, 'validate_add_to_cart' ), 10, 1 );
+
+		if ( function_exists( 'is_checkout' ) && is_checkout() && self::cart_contains_change_address_request() ) {
+			self::attach_callbacks_for_address_change_request();
+		}
+
+		add_filter( 'the_title', array( __CLASS__, 'title_for_address_change' ), 100 );
+
+		// Applying coupons via this flow is not supported.
+		add_filter( 'woocommerce_coupons_enabled', '__return_false' );
+
+		// Add breadcrumbs for the address change request.
+		add_filter( 'woocommerce_get_breadcrumb', array( __CLASS__, 'crumbs_for_address_change' ), 10, 1 );
+
+		// Cart
+		self::$edit_subscription_cart_instance = new WCS_Cart_Change_Address();
```

Comment:
Hi @james-allan ! It looks like this class does not exist anymore. Do you know if it were renamed or replaced?

Reply:
It was a new file I created while working on this but I never submitted the file. I've just pushed it up in [aff4ebb](https://github.com/Automattic/woocommerce-subscriptions-core/pull/219/commits/aff4ebb3a8ce9808873ed91a556806d90d1bb819).

It's very much a WIP like the rest of this PR. :)